apps/common/src/main/scala/org/lfdecentralizedtrust/splice/util/SpliceCircuitBreaker.scala

```diff
@@ -47,11 +76,27 @@ class SpliceCircuitBreaker(name: String, underlying: CircuitBreaker)(implicit
     }
   }
 
-  private def callAndMark[T](body: => Future[T]) = {
+  private def callAndMark[T](body: => Future[T])(implicit tc: TraceContext) = {
+    lastFailure.updateAndGet(_.filter { lastFailureTime =>
+      val elapsed = clock.now - lastFailureTime
+      if (elapsed.compareTo(config.resetFailuresAfter.asJava) >= 0) {
```

Comment:
won't this mean that we will always close the breaker when `resetTimeout` exceeds `resetFailuresAfter`?